eclass/dist-kernel-utils.eclass

```diff
@@ -66,43 +66,38 @@ dist-kernel_get_image_path() {
 }
 
 # @FUNCTION: dist-kernel_install_kernel
-# @USAGE: <version> <image> <system.map>
+# @USAGE: <kernel_dir>
 # @DESCRIPTION:
-# Install kernel using installkernel tool.  <version> specifies
-# the kernel version, <image> full path to the image, <system.map>
-# full path to System.map.
+# Install kernel using installkernel tool. kernel_dir specifies the full
+# path to the installed kernel directory to use.
 dist-kernel_install_kernel() {
 	debug-print-function ${FUNCNAME} "${@}"
 
-	[[ ${#} -eq 3 ]] || die "${FUNCNAME}: invalid arguments"
-	local version=${1}
-	local image=${2}
-	local map=${3}
+	[[ ${#} -eq 1 ]] || die "${FUNCNAME}: invalid arguments"
+	local kernel_dir=${1}
 
 	ebegin "Installing the kernel via installkernel"
-	# note: .config is taken relatively to System.map;
-	# initrd relatively to bzImage
-	ARCH=$(tc-arch-kernel) installkernel "${version}" "${image}" "${map}"
+	pushd ${kernel_dir} >/dev/null || die "Invalid kernel dir: ${kernel_dir}"
+	ARCH=$(tc-arch-kernel) emake install
```

Comment:
NAK. That's hardly an improvement. You're relying on the `install` target having no side effects. Also, the kernel Makefiles hardcode paths, so fix that first.

Reply:
> NAK. That's hardly an improvement. You're relying on the `install` target having no side effects. Also, the kernel Makefiles hardcode paths, so fix that first.

Right, I hadn't thought about the hardcoded path, that is a good point. I've dropped this commit, leaving just the part that drops the code paths for installkernel-gentoo-7.  I don't really care about being able to set `INSTALLKERNEL` to some alternative script anyway, just thought it was a nice to have, but if it is going to require patching the kernel makefiles then it's not worth the effort I think.